### PR TITLE
Fix unicode cache keys on Python 2.7

### PR DESCRIPTION
**Pull request by linux-warrior**

### CODE REVIEW

**Comment by encbladexp** [comment]
@niwinz is it planned to merge this? We have also this issue with Unicode Keys.

**Comment by jordiecometrica** [comment]
I would like this fix too. I can work around it for now in Python 2 by making sure I encode `str` before setting a cache key.

File: django_redis/util.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 
 @python_2_unicode_compatible
-class CacheKey(str):
+class CacheKey(object):
```
Comment (jordiecometrica) [inline django_redis/util.py]: This fix seems wrong. A cache key should be a string. You should be able to call `len()`, `format`, and other stringy things on it. Doesn't this change break that?

Comment (linux-warrior) [inline django_redis/util.py]: Even though `CacheKey` is no longer a string, it still has a `__str__` method (or `__unicode__` on Python 2). When we use an instance of this class as an argument for a Redis command, this method will convert it to the contained cache key. And why do you need to call `len()` or `format` on a cache key wrapper?

**Comment by niwinz** [comment]
Hi @linux-warrior, please keep the PR small and focused. I don't understand the need to add a new dependency, is that mandatory for the fix?

**Comment by linux-warrior** [comment]
The problem with the current code in `django_redis.client.sharded` and some other places is the usage of the `str` class, which has different meanings on Python 2 and Python 3. The usual way to resolve this and similar problems is to use a Python 2 and 3 compatibility library like `six` or `future`. Another popular option is to create a `compat.py` file in the project and define a `text_type` wrapper there, among other things. What would you recommend doing?

**Comment by niwinz** [comment]
I prefer keep the things simple and not adding an additional dependency. If this is just one thing, just define a compat.py module and there the need code.

Thanks

**Comment by linux-warrior** [comment]
Done.

**Comment by jordiecometrica** [comment]
@niwinz Sorry to be a bother, but will you be doing a new release of django-redis soon? If not, I can just point my requirements.txt to the git repo instead of pypi.

**Pull request merged**